spark/src/main/scala/org/apache/comet/rules/CometScanRule.scala

```diff
@@ -261,6 +261,10 @@ case class CometScanRule(session: SparkSession) extends Rule[SparkPlan] {
 
     val fallbackReasons = new ListBuffer[String]()
 
+    if (CometSparkSessionExtensions.isSpark40Plus) {
+      fallbackReasons += s"$SCAN_NATIVE_ICEBERG_COMPAT  is not implemented for Spark 4.0.0"
+    }
```

Comment:
We can revisit this after https://github.com/apache/datafusion-comet/pull/1830 is merged